vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/ExecutionPlanBuilder.java

```diff
@@ -84,6 +86,25 @@
     @Nonnull
     ExecutionPlanBuilder with(@Nonnull ProgressTrackerListener listener);
 
+    
+    /**
+     * Sets packages handled externally ahead of execution for prevalidation of plan
+     * @param List of packages handled by other builder
+     * @return this.
+     */
+    @Nonnull
+    ExecutionPlanBuilder with(@Nonnull List<PackageId> externalPackages);
+
+    /**
+     * Triggers Validation and returns PackageIds of all packages to be installed
+     * by this builder
+     * @return List of packages to be installed by this builder.
+     * @throws IOException if an I/O error occurrs.
+     * @throws PackageException if the plan is not valid.
+     */
+    @Nonnull
+    List<PackageId> calculateIds() throws IOException, PackageException;
```

Comment:
- Should be a Set (unless the order is important)
- I don't like the name. The tasks could also be removal tasks. so what about `List<PackageTask> preview()` ?

Reply:
fixed